templates/react-native/.github/workflows/publish.yml.twig

```diff
@@ -0,0 +1,32 @@
+name: Publish to NPM
+
+on:
+  release:
+    types: [published]
+  workflow_dispatch:
+    
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    # Setup Node.js environment
+    - name: Use Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: '20.x'
+        registry-url: 'https://registry.npmjs.org'
+
+    - name: Install dependencies and build
+      run: |
+        npm install
+        npm run build
+
+    # Publish to NPM
+    - name: Publish
+      run: npm publish
+      env:
+        NODE_AUTH_TOKEN: ${{'{{'}}secrets.NPM_TOKEN{{'}}'}}
```

Comment:
We need to be able to differentiate a RC release vs the latest release. Maybe we need two different workflows where one does [`npm publish --tag next`](https://docs.npmjs.com/cli/v8/commands/npm-publish#tag)?

Reply:
Do we do something similar for web?

Reply:
Yep: https://github.com/appwrite/sdk-for-web/blob/e907a873fa53452e4c746ac8afbfb3ee07637958/.travis.yml#L18

Reply:
Updated the workflow